compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp

```diff
@@ -1952,3 +1952,14 @@ extern "C" int32_t LLVMRustGetElementTypeArgIndex(LLVMValueRef CallSite) {
 extern "C" bool LLVMRustIsBitcode(char *ptr, size_t len) {
   return identify_magic(StringRef(ptr, len)) == file_magic::bitcode;
 }
+
+extern "C" bool LLVMRustIsNonGVFunctionPointerTy(LLVMValueRef V) {
+  if (unwrap<Value>(V)->getType()->isPointerTy()) {
+    if (auto *GV = dyn_cast<GlobalValue>(unwrap<Value>(V))) {
+      if (GV->getValueType()->isFunctionTy())
+        return false;
+    }
+    return true;
```

Comment:
Should the `return true` be moved up one line? The name of this function is also a little confusing to me. The behavior seems to be "true for any pointer type except pointers to functions" and the name reads to me like it's the opposite of that.

Reply:
This function tests if a LLVM value is of pointer type and its value isn't a _global value_ function type (i.e., it returns true if it is (a) a pointer type and (b) its value is not a _global value_ function type). We're only interested in  indirect branch targets (i.e., targets that are not known at code generation). Does it make sense to you?

Reply:
I guess I'm not sure what the difference between a "global value function type" and a "function pointer" is. Could you explain that more? Is this the difference between a pointer to a specific function vs a pointer an arbitrary function?

Reply:
A LLVM value of pointer type where its value is a *global value* function type as a branch target means that the target destination (address) is known at code generation, resulting in a direct branch, and also causing the error you see in #90546 when used as an argument to the llvm.type.test. Does it make sense?